### PR TITLE
Upgrade keras 3

### PR DESCRIPTION
**Pull request by calderjo**
this is based on [original rosbo pr for keras-3](https://github.com/Kaggle/docker-python/pull/1357) to upgrade to keras 3

key things:
- a patch was added ([commit](https://github.com/tensorflow/decision-forests/commit/b9a4cf110b055c41c48286a2484ba1b3750533c9?diff=split&w=0)) to address unreleased version of tf-df (short term in-order to unblock upcoming project)
- tensorflowjs along with its test was removed 

http://b/318672158
passes:
https://ci.kaggle.net/job/kernels/job/docker-python/job/upgrade-keras-3/

cc: @jplotts

### CODE REVIEW

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -202,18 +202,63 @@ RUN apt-get install -y default-jre && \
...
     /tmp/clean-layer.sh
 
-# TODO(b/318672158): Upgrade to Keras 3 once compatible with other TF libries.
-# See blockers here: https://b.corp.google.com/issues/319722433#comment8
-RUN pip install keras keras-cv keras-nlp && \
+# b/318672158 Use simply tensorflow_decision_forests on next release, expected with tf 2.16
+RUN pip install tensorflow_decision_forests --no-deps && \
+    /tmp/clean-layer.sh
+
+RUN sed -i "/import tensorflow_decision_forests as tfdf/a import tf_keras" /opt/conda/lib/python3.10/site-packages/tensorflow_decision_forests/__init__.py && \
```
[the remark this answers is not in the document]
Comment (djherbis) [reply Dockerfile.tmpl:219]: haha wow, this was more complex than I was hoping, but amazing job patching this much.

Can we move this series of sed commands into a file, maybe keras_patch.sh and call it here?

That would help make it less complex inside the Dockerfile itself.

[the remark this answers is not in the document]
Comment (calderjo) [reply Dockerfile.tmpl]: your wish is my command. d.

**Comment by calderjo** [comment]
btw had some trouble figuring out git rebase via cli, create a new different branch and forgot to remove the tfjs test, which caused it to failed. it been removed we should expect tests to pass, the o.g branch did pass.

**Review by djherbis — APPROVED** [review]
Amazing work mate, super cool to see this build/pass tests.

**Pull request merged**